project/ws/app/src/lib/routes/profile-v2/components/verify-otp/verify-otp.component.ts

```diff
@@ -0,0 +1,106 @@
+import { Component, OnInit, Inject, OnDestroy, ViewChild, Output, EventEmitter } from '@angular/core'
+import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog'
+import { HttpErrorResponse } from '@angular/common/http'
+import { MatSnackBar } from '@angular/material'
+
+import { Subject } from 'rxjs'
+import { takeUntil } from 'rxjs/operators'
+/* tslint:disable */
+import * as _ from 'lodash'
+
+import { OtpService } from '../../../user-profile/services/otp.services'
+
+@Component({
+  selector: 'ws-verify-otp',
+  templateUrl: './verify-otp.component.html',
+  styleUrls: ['./verify-otp.component.scss'],
+})
+
+export class VerifyOtpComponent implements OnInit, OnDestroy {
+
+  private destroySubject$ = new Subject()
+  @ViewChild('timerDiv', { static: false }) timerDiv !: any
+  @Output() resendOTP = new EventEmitter<string>()
+  @Output() otpVerified = new EventEmitter<string>()
+  timeLeft = 90
+  interval: any
+  showResendOTP = false
+  otpEntered = ''
+  constructor(
+    public dialogRef: MatDialogRef<VerifyOtpComponent>,
+    @Inject(MAT_DIALOG_DATA) public data: any,
+    private otpService: OtpService,
+    private matSnackbar: MatSnackBar
+  ) { }
+
+  ngOnInit() {
+    this.startTimer()
+  }
+
+  startTimer() {
+    this.interval = setInterval(() => {
+      if (this.timeLeft > 0) {
+        this.timeLeft = this.timeLeft - 1
+        this.timerDiv.nativeElement.innerHTML = `${Math.floor(this.timeLeft / 60)}m: ${this.timeLeft % 60}s`
```

Comment:
hardcoded value should come from constant

Reply:
same as above